declarations.d.ts

```diff
@@ -403,22 +403,37 @@ interface IHook {
 /**
  * Describes TypeScript compilation methods.
  */
-interface ITypeScriptCompilationService {
+interface ITypeScriptService {
 	/**
-	 * Compiles specified files only. Options are read from CLI's config file. Disregards tsconfig.json.
-	 * @param {any} compilerOptions: Specifies if noEmitOnError option is true or false. When it is true, any warning will prevent .js generation.
-	 * @param {string[]} typeScriptFiles The files that will be compiled.
+	 * Transpiles specified files or all files in the project directory. The default passed options are overriden by the ones in tsconfig.json file. The options from tsconfig.json file are overriden by the passed compiler options.
+	 * @param {string} projectDir: Specifies the directory of the project.
+	 * @param {string[]} typeScriptFiles @optional The files that will be compiled.
 	 * @param {string[]} definitionFiles @optional The definition files used for compilation.
-	 * @return {IFuture<void>}
+	 * @param {ITypeScriptTranspileOptions} options @optional The transpilation options.
+	 * @return {IFuture<string>} The result from the TypeScript transpilation.
 	 */
-	compileFiles(compilerOptions: { noEmitOnError: boolean }, typeScriptFiles: string[], definitionFiles?: string[]): IFuture<void>;
+	transpile(projectDir: string, typeScriptFiles?: string[], definitionFiles?: string[], options?: ITypeScriptTranspileOptions): IFuture<string>;
 
 	/**
 	 * Spawns tsc directly without options. Tsc will respect tsconfig.json file in case it exists and all of its options.
-	 * @param {any} compilerOptions: Specifies if noEmitOnError option is true or false. When it is true, any warning will prevent .js generation.
-	 * @return {IFuture<void>}
+	 * @param {ITypeScriptTranspileOptions} @optional options The transpilation options.
+	 * @return {IFuture<string>} The result from the TypeScript transpilation.
+	 */
+	transpileWithDefaultOptions(projectDir: string, options?: ITypeScriptTranspileOptions): IFuture<string>;
+
+	/**
+	 * Returns new object, containing all typeScript and all TypeScript definition files.
```

Comment:
Minor typo - the first letter of the first occurrence of `TypeScript` is not capitalized on this line